### PR TITLE
refactor(mesh): replace use-auth-state with user context

### PR DESCRIPTION
**Pull request by caesarWHLee**
# Notable change
Fetch user in root layout and then pass to the user context. 
Expand user schema to store cross-page data including `picks`, `followingMembers`, `followingCategories` and `followingPublishers`.

## Replace `use-auth-state` hook
Each component using `use-auth-state` will need to request for the user data which is available in server component. Add user context to wrap all pages so the user data can be used by every client componets.

## Follow/Pick button update
Each button with `followingMember`/`storyId` prop can now work with user context to know if the member is followed or story is picked.

## Follow-up
1. mutating user is quite cumbersome and low performance: update context with redux so the mutation could be define with reducers and the mutation won't cause all the components re-render. And new slices could be added to fit the needs.
2. sync user info: update user data when user logout or token expired.
3. performance concern (future): Since an user could picks lots of stories or follow lots of members, the approach now could delay the user gql response. A member with 3000 picks could take approximately 1 more seconds  to get the response which should be fine for now.

### CODE REVIEW

**Comment by caesarWHLee** [comment]
關於 Mesh client side 相關的改動，請大家幫忙看一下有沒有需要調整的地方

File: packages/mesh/components/story-card/story-pick-button.tsx [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
-  //TODO: replace with logged-in member's status
-  const params = useParams()
-  const { id } = params
-  const memberId = Array.isArray(id) ? id[0] : id
+export default function StoryPickButton({ storyId }: { storyId: string }) {
+  const router = useRouter()
+  const { user, setUser } = useUser()
+  const memberId = user?.memberId
+  const isStoryPicked =
+    user?.picks.findIndex((pickInfo) => pickInfo.story?.id === storyId) !== -1
```
Comment (tyc67) [inline packages/mesh/components/story-card/story-pick-button.tsx]: 提問: 如果是存 `Set` / `Map` ids, 取代 `findIndex`, `filter`, Component 使用起來會比較好嗎?

Comment (caesarWHLee) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 目前的需求用 Set / Map 比較方便，我再加個 transform function 轉一下 gql 的資料好了

File: packages/mesh/app/story/[id]/_components/article.tsx [hunk 2]
```diff
@@ -81,7 +81,7 @@ export default function Article({
             <Image
               src={story.og_image}
               alt="hero image"
-              objectFit="cover"
+              style={{ objectFit: 'cover' }}
```
Comment (erase2004) [inline packages/mesh/app/story/[id]/_components/article.tsx:84]: 提問，這兩者實際上會有什麼差別？ :raising_hand:

**Review by erase2004 — APPROVED** [review]
LGTM :heavy_check_mark:

**Review by LinHeMa — APPROVED** [review]
LGTM👍

**Comment by LinHeMa** [comment]
順帶提問：
現在如果更改使用者資料，需要同時發送 `apollo mutate` 以及更改 `context` 裡面的內容？
然後 `client component` 主要的 `state` 是從 `context` 裡面拿？

File: packages/mesh/app/story/[id]/_components/article.tsx (hunk 2 above)
Comment (caesarWHLee) [reply packages/mesh/app/story/[id]/_components/article.tsx:84]: 這個其實是在 `next dev` 執行的時候跳出的 warning，看來在 [next/image 13](https://nextjs.org/docs/messages/next-image-upgrade-to-13) 之後因為支援了 `style` prop，所以把以前的作法設成 depreacted 了

```
Image with src "https://statics-readr-tw-prod.readr.tw/images/e6d20324-ab48-4e5f-8ba7-03dd12352045-w800.jpg" has legacy prop "objectFit". Did you forget to run the codemod?
Read more: https://nextjs.org/docs/messages/next-image-upgrade-to-13
```

**Comment by caesarWHLee** [comment]
> 順帶提問： 現在如果更改使用者資料，需要同時發送 `apollo mutate` 以及更改 `context` 裡面的內容？ 然後 `client component` 主要的 `state` 是從 `context` 裡面拿？

對，基本上就是 client side maintain 一份 user state，然後勁量做到跟 cms 中的資料一致，至於 UX 上面的行為可能有定義的空間，像是這個 PR 中使用者精選的時候會預設先精選成功，如果精選的 request 遇到錯誤的話會再把已精選改回沒精選，主要是優化點擊後精選的反應速度讓使用者不用等待 response，但這種調教就看大家遇到覺得怎麼處理比較好，有討論空間

File: packages/mesh/components/story-card/story-pick-button.tsx (hunk 1 above)
Comment (caesarWHLee) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 已更新在 [7a5a649](https://github.com/readr-media/Sachiel/pull/826/commits/7a5a649c085f69ed2d9d5b7a1f8d44e058c7afe8)

File: packages/mesh/app/story/[id]/_components/article.tsx [hunk 3]
```diff
@@ -81,7 +81,8 @@ export default function Article({
             <Image
               src={story.og_image}
               alt="hero image"
-              style={{ objectFit: 'cover' }}
+              objectFit="cover"
+              // style={{ objectFit: 'cover' }}
```
Comment (tyc67) [inline packages/mesh/app/story/[id]/_components/article.tsx]: 你這邊不是要用 `style` props 嗎?

Comment (caesarWHLee) [reply packages/mesh/app/story/[id]/_components/article.tsx]: 要印 log 貼上來結果忘記改回來 😓 ，修正在 [11f36d7](https://github.com/readr-media/Sachiel/pull/826/commits/11f36d7648e1d9e30612492161f7fbabc2976a73)

**Review by tyc67 — APPROVED** [review]

**Pull request merged**